lib/Target/LLVMIR/LLVMIRTranslation.cpp

```diff
@@ -189,6 +189,11 @@ translateTritonGPUToLLVMIR(llvm::LLVMContext *llvmContext,
 
   llvm::SMDiagnostic err;
   for (auto &lib : extern_libs) {
+    if ("libdevice" == lib.first) {
+      linkLibdevice(*llvmir, lib.first);
```

Comment:
Minor issue. I think the following code (192-204) is somehow redundant with the code in `linkLibdevice`, we may want to refactor it later.

Reply:
How about sperate link with opt?

Reply:
Yeah, it would be nice if you could have a more general suggestion, but not a requirement.